gdal/frmts/wcs/wcsutils.cpp

```diff
@@ -0,0 +1,698 @@
+                              CPLXMLNode *node,
+                              CPLString &path,
+                              CPLString from,
+                              std::vector<CPLString> keys)
+{
+    CPLXMLNode *node2 = CPLGetXMLNode(node, from);
+    if (node2) {
+        path = path + from + ".";
+        for (unsigned int i = 0; i < keys.size(); i++) {
+            if (CPLXMLNode *node3 = CPLGetXMLNode(node2, keys[i])) {
+                CPLString name = path + keys[i];
+                CPLString value = CPLGetXMLValue(node3, NULL, "");
+                value.Trim();
+                *metadata = CSLSetNameValue(*metadata, name, value);
+            }
+        }
+    }
+    return node2;
+}
+
+CPLString GetKeywords(CPLXMLNode *root,
+                      CPLString path,
+                      CPLString kw)
+{
+    CPLString words = "";
+    CPLXMLNode *keywords = (path != "") ? CPLGetXMLNode(root, path) : root;
+    if (keywords) {
+        std::vector<unsigned int> epsg_codes;
+        for (CPLXMLNode *node = keywords->psChild; node != NULL; node = node->psNext) {
+            if (node->eType != CXT_Element) {
+                continue;
+            }
+            if (kw == node->pszValue) {
+                CPLString word = CPLGetXMLValue(node, NULL, "");
+                word.Trim();
+
+                // crs, replace "http://www.opengis.net/def/crs/EPSG/0/" with EPSG:
+                const char *epsg = "http://www.opengis.net/def/crs/EPSG/0/";
+                size_t pos = word.find(epsg);
+                if (pos == 0) {
+                    CPLString code = word.substr(strlen(epsg), std::string::npos);
+                    if (code.find_first_not_of(DIGITS) == std::string::npos) {
+                        epsg_codes.push_back(atoi(code));
+                        continue;
+                    }
+                }
+
+                // profiles, remove http://www.opengis.net/spec/
+                // interpolation, remove http://www.opengis.net/def/interpolation/OGC/1/
+
+                const char *spec[] = {
+                    "http://www.opengis.net/spec/",
+                    "http://www.opengis.net/def/interpolation/OGC/1/"
+                };
+                for (unsigned int i = 0; i < sizeof(spec)/sizeof(spec[0]); i++) {
+                    pos = word.find(spec[i]);
+                    if (pos != std::string::npos) {
+                        word.erase(pos, strlen(spec[i]));
+                    }
+                }
+
+                if (words != "") {
+                    words += ",";
+                }
+                words += word;
+            }
+        }
+        if (epsg_codes.size() > 0) {
+            CPLString codes;
+            std::sort(epsg_codes.begin(), epsg_codes.end());
+            unsigned int pajazzo = 0, i = 0, a = 0, b = 0;
+            while (1) {
+                unsigned int c = i < epsg_codes.size() ? epsg_codes[i] : 0;
+                if (pajazzo == 1) {
```

Comment:
pajazzo ?

Reply:
Pajazzo (pajatso) is (was) a popular game machine played with coins in Finland. Here we keep on feeding the loop with EPSG codes (coins) and some are saved and some are lost. Greatly reduces the long list given by GeoServer. And many coins are lost. :) Pajatso is also the Finnish name for an Italian opera Pagliacci.